husky_directory/services/pws.py

```diff
@@ -236,6 +239,26 @@ def list_persons(self, request_input: ListPersonsInput) -> ListPersonsOutput:
         payload = request_input.payload
         constraints = self.global_constraints + request_input.constraints
 
+        # These constraints can only be created when a request is made, because
+        # this method only knows about one population at a time, as each
+        # list_persons query can only target one "OR" population.
+        student_population_constraint = RecordConstraint(
+            namespace="PersonAffiliations.StudentPersonAffiliation",
+            predicate=lambda v: "students" in populations,
+            failure_callback=partial(
+                clear_namespace, "PersonAffiliations.StudentPersonAffiliation"
+            ),
+        )
+        employee_population_constraint = RecordConstraint(
+            namespace="PersonAffiliations.EmployeePersonAffiliation",
+            predicate=lambda v: "employees" in populations,
+            failure_callback=partial(
+                clear_namespace, "PersonAffiliations.EmployeePersonAffiliation"
+            ),
+        )
+        constraints.insert(0, student_population_constraint)
+        constraints.insert(0, employee_population_constraint)
```

Comment:
Prepend these to the global constraints, so that we immediately trash records that don't match our needs (saves subsequent processing)